packages/core/useFetch/index.ts

```diff
@@ -0,0 +1,161 @@
+   */
+  abort: Fn
+
+  /**
+   * Manually call the fetch
+   */
+  execute: Fn
+}
+
+interface UseFetchOptions {
+  /**
+   * Will automatically run fetch when `useFetch` is used
+   *
+   * @default true
+   */
+  immediate?: boolean
+
+  /**
+   * Will automatically refetch when the URL is changed if the url is a ref
+   *
+   * @default false
+   */
+  refetch?: MaybeRef<boolean>
+}
+
+export function useFetch(url: MaybeRef<string>): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, useFetchOptions: UseFetchOptions): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, options: RequestInit, useFetchOptions?: UseFetchOptions): UseFetchReturn
+
+export function useFetch(url: MaybeRef<string>, ...args: any[]): UseFetchReturn {
+  const supportsAbort = typeof AbortController === 'function'
+
+  let fetchOptions: RequestInit = {}
+  let options: UseFetchOptions = { immediate: true, refetch: false }
+
+  if (args.length > 0) {
+    if ('immediate' in args[0] || 'refetch' in args[0])
+      options = { ...options, ...args[0] }
+    else
+      fetchOptions = args[0]
+  }
+
+  if (args.length > 1) {
+    if ('immediate' in args[1] || 'refetch' in args[1])
+      options = { ...options, ...args[1] }
+  }
+
+  const isFinished = ref(false)
+  const isFetching = ref(false)
+  const status = ref<number | null>(null)
+  const response = ref<Response | null>(null)
+  const error = ref<any>(null)
+  const data = ref<string | object | null>(null)
+
+  const canAbort = computed(() => {
+    return supportsAbort && isFetching.value
+  })
+
+  let controller: AbortController | null
+
+  const abort = () => {
+    if (supportsAbort && controller instanceof AbortController)
+      controller.abort()
+  }
+
+  const execute = () => {
+    isFetching.value = true
+    isFinished.value = false
+    error.value = null
+
+    if (supportsAbort) {
+      controller = new AbortController()
+      fetchOptions = { ...fetchOptions, signal: controller.signal }
+    }
+
+    fetch(unref(url), fetchOptions)
+      .then((fetchResponse) => {
+        response.value = fetchResponse
+        status.value = fetchResponse.status
+
+        const contentType = fetchResponse.headers.get('content-type')
+
+        if (contentType && contentType.includes('application/json'))
+          fetchResponse.json().then(json => data.value = json)
+        else
+          fetchResponse.text().then(text => data.value = text)
+
+        // see: https://www.tjvantoll.com/2015/09/13/fetch-and-errors/
+        if (!fetchResponse.ok)
+          throw new Error(fetchResponse.statusText)
+      })
+      .catch(fetchError => error.value = fetchError.message)
+      .finally(() => {
+        isFinished.value = true
+        isFetching.value = false
+      })
+  }
+
+  if (isRef(url)) {
+    if (isRef(options.refetch))
+      watch([options.refetch, url], () => unref(options.refetch) && execute())
+    else if (options.refetch)
+      watch(url, () => execute())
+  }
```

Comment:
```suggestion
watch(
()=>{
unref(url)
unref(options.refetch)
}, 
() => unref(options.refetch) && execute()
)
```

Sorry for the formatting

Reply:
No problem, I can fix it in a sec

Reply:
This doesn't actually seem to work. Can you watch an `unref`? I was under the impression it just gave you back the raw value if it was a ref

Reply:
Did you test it? It's not watching an `unref` but the "accessing". `unref` is just a unified way to accessing here, the return value doesn't matter. Should work I think.

Reply:
Ok it looks like it does work, but the watch needs to have `deep: true` set

Reply:
Em? I don't think it's needed... 🤔

Reply:
Weird, it only seems to work when I set `deep: true`. Not really sure why

Reply:
I just created a repo to demonstrate the issue. It is a bit of a contrived example, but it should show the issue. It is in vite so you should be able to just clone it and run it